src/server/force-refresh.integration.test.ts

```diff
@@ -0,0 +1,472 @@
+      .spyOn(Auth0Client.prototype as any, "saveToSession")
+      .mockResolvedValue(undefined);
+
+    // --- Execution ---
+    const auth0Client = new Auth0Client({
+      domain: DEFAULT.domain,
+      clientId: DEFAULT.clientId,
+      clientSecret: DEFAULT.clientSecret,
+      appBaseUrl: DEFAULT.appBaseUrl,
+      secret: secret
+    });
+
+    const mockReq = new NextRequest("https://example.com/api/pages-test");
+    const mockRes = new NextResponse();
+
+    // Use expected values from beforeEach setup
+    const expectedRefreshedAccessToken =
+      "getAccessToken-refreshed-access-token";
+    const expectedRefreshedRefreshToken =
+      "getAccessToken-refreshed-refresh-token";
+    const expectedRefreshedExpiresIn = 7200;
+
+    const result = await auth0Client.getAccessToken(mockReq, mockRes, {
+      refresh: true
+    });
+
+    // --- Assertions ---
+
+    // 1. Assert the returned access token details
+    expect(result).not.toBeNull();
+    expect(result?.token).toBe(expectedRefreshedAccessToken);
+    expect(result?.expiresAt).toBeGreaterThan(initialExpiresAt);
+    const expectedExpiresAtRough =
+      Math.floor(Date.now() / 1000) + expectedRefreshedExpiresIn;
+    expect(result?.expiresAt).toBeGreaterThanOrEqual(
+      expectedExpiresAtRough - 5
+    );
+    expect(result?.expiresAt).toBeLessThanOrEqual(expectedExpiresAtRough + 5);
+    expect(result?.scope).toBe(initialTokenSet.scope); // Scope remains initial
+
+    // 2. Assert saveToSession was called (Pages Router specific)
+    expect(mockSaveToSession).toHaveBeenCalledOnce();
+    const savedSession = mockSaveToSession.mock.calls[0][0] as SessionData;
+    expect(savedSession.tokenSet.accessToken).toBe(
+      expectedRefreshedAccessToken
+    );
+    expect(savedSession.tokenSet.refreshToken).toBe(
+      expectedRefreshedRefreshToken
+    );
+    expect(savedSession.tokenSet.expiresAt).toBe(result?.expiresAt);
+    expect(savedSession.tokenSet.scope).toBe(initialTokenSet.scope);
+
+    // 3. Assert mockFetch was called
+    const fetchCalls = mockFetch.mock.calls;
+    expect(fetchCalls.length).toBeGreaterThan(0);
+    const tokenEndpointCall = fetchCalls.find((call: any) => {
+      let urlString: string;
+      if (call[0] instanceof URL) {
+        urlString = call[0].toString();
+      } else if (call[0] instanceof Request) {
+        urlString = call[0].url;
+      } else {
+        urlString = call[0];
+      }
+      return urlString.endsWith("/oauth/token");
+    });
+    expect(tokenEndpointCall).toBeDefined();
+  });
+
+  /**
+   * Test Case: App Router Overload - getAccessToken(options)
+   * Verifies that when called without req/res objects and refresh: true,
+   * it refreshes the token. Currently, it *also* calls saveToSession,
+   * so the test asserts this observed behavior.
+   */
+  it("should refresh token for app-router overload when refresh is true", async () => {
+    const initialExpiresAt = Math.floor(Date.now() / 1000) - 60;
```

Comment:
Why is the token expired in the past, when this test is explicitly to test non-expired token refresh?